### PR TITLE
#182 Order Callback | Interval value not stored in sales_order_item and not shown in order email. 

### PR DESCRIPTION
**Pull request by eugene-shab**
feat: Fix interval value for callback orders.

Refs: #182

### CODE REVIEW

File: Model/Quote/SubscriptionOption/Updater.php [hunk 1]
```diff
@@ -54,7 +54,7 @@ public function update(
             $createNewSubscriptionAtCheckout = true;
         }
 
-        if (!$createNewSubscriptionAtCheckout) {
+        if (!$createNewSubscriptionAtCheckout && !$this->quoteItemHelper->isItemFulfilsSubscription($quoteItem)) {
```
Comment (garthbrantley) [inline Model/Quote/SubscriptionOption/Updater.php:57]: Please explain why this is needed?

**Comment by eugene-shab** [comment]
## Platform
i add product to the order callback in this way.
i've tried different order type also.



## From the Magento side
Our code:
Screenshot from 2023-11-30 17-14-35

#### Case, Fulfils order callback:
Request comes into this file `Updater::update()`. Originally (before fix) in `update()` we have a condition like:
```
        if (!$createNewSubscriptionAtCheckout) {
            $subscriptionInterval = null;
        }
```
in row `50` i see a `$subscriptionOption = 'onetime_purchase'`
after that on row `57-58` clean up the variable`$subscriptionInterval`, this caused a problem for Fulfils order request.


#### Case, adding subscription product with Interval Weekly from frontend.
Request comes to updater and `$subscriptionOption = 'subscription'
so, on the row `50` code met condition and set variable `$createNewSubscriptionOnCheckout = true`,
The interval was not cleaned.


#### Case, adding `One Time Delivery` product from frontend.
The Request comes to updater and `$subscriptionOption = 'onetime_purchase'`, on line `50` the variable doesn't met a condition.
The interval will be cleaned.

**Comment by garthbrantley** [comment]
@eugene-shab What about the case of a recurring order placed by SP via the M2 API?

**Comment by eugene-shab** [comment]
> @eugene-shab What about the case of a recurring order placed by SP via the M2 API?

@garthbrantley As I can see recurring order placed by SP via the M2 API working by another callstack and saving `Interval`

```
{
    "subscription_option": {
        "is_fulfilling": true,
        "subscription_id": 4790338,
        "interval": "Weekly",
        "reorder_ordinal": "1",
        "item_added_by_subscribe_pro": true,
        "item_fulfils_subscription": true,
        "next_order_date": "2023-12-07"
    },
    "qty": 1
}
```
`Updater::update()` didn't call

**Pull request merged**